en/security.rst

```diff
@@ -18,8 +18,298 @@ ACL (Access Control List)
+
+.. code-block:: sql
+
+	CREATE TABLE tde_tbl (att1 INT, att2 VARCHAR(20)) ENCRYPT=AES;
+
+When an encrypted table is created, all data related to the table is automatically encrypted when written to disk, and decrypted when read into memory. Related data includes not only tables, but also indexes created on the table, temporary data created while executing queries related to the table, logs created when data is changed, DWB, and backups. For more details, see :ref:`tde-enc-target` and :ref:`tde-restriction`.
+
+.. _tde-key:
+
+Key Management
+--------------
+
+CUBRID uses symmetric key algorithms to encrypt the data. Keys used for encryption are managed in two levels consisting of master keys and data keys for efficiency. Master keys managed by the user is stored in a separate file, and CUBRID provides a utility to manage it.
+
+.. _tde-2level-key:
+
+2-Level Key Management
+^^^^^^^^^^^^^^^^^^^^^^
+
+CUBRID TDE manages keys in two levels as follows:
+
+.. image:: /images/tde_2_level_key.png
+  :width: 469
+  :align: center
+  :alt: 2 Level Key Management Image
+
+*    **Master key**: A key used when encrypting and decrypting data keys, and is managed by DBA user.
+*    **Data Key**: A key used when encrypting user data such as table and log, and is managed by CUBRID Engine.
+
+Data keys are stored within the data volume and are always securely encrypted using a master key when written to disk. The master key is stored in a separate file and it must be managed safely according to a security policy users comply with.
+
+Managing keys in two levels makes it possible to efficiently perform the key change operation. If there is only a key that encrypts the user data, when you change the key, it takes a long time to work because all the data which has been encrypted has to be read, and decrypted and re-encrypted. Also, the overall performance of the database may be degraded during this process.
+
+.. warning:: **Loss of Master Key**
+    
+    If the master key is lost, data encrypted by TDE cannot be read or changed.
+
+.. _tde-file-based-key: 
+
+File-based Master Key Management
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Master keys are separately stored and managed as a separate key file so that the user can manage master keys in various ways according to individual security requirements. This key file contains all the information of master keys, so if it is leaked, there may be a security problem, and if it is lost, the encrypted data cannot be read (:ref:`tde-load-failure`). So, be careful to manage this key file.
+
+By default, the key file is created with the name of **<database-name>_keys** at the location where the data volume is created when creating a database using **cubrid createdb** utility. Without additional configuration for the key file, this key file is automatically used. The location of the key file te be used can be changed by a system parameter. For more information, see :ref:`disk-parameters`.
```

Comment:
```suggestion
By default, the key file is created with the name of **<database-name>_keys** at the location where the data volume is created when creating a database using **cubrid createdb** utility. Without additional configuration for the key file, this key file is automatically used. The location of the key file can be changed by a system parameter. For more information, see :ref:`disk-parameters`.
```
is it right?

Reply:
```suggestion
By default, the key file is created with the name of **<database-name>_keys** at the location where the data volume is created when creating a database using **cubrid createdb** utility. Without additional configuration for the key file, this key file is automatically used. The location of the key file to be used can be changed by a system parameter. For more information, see :ref:`disk-parameters`.
```
This may be correct.